packages/amplify-codegen-model-plugin/src/languages/swift-declaration-block.ts

```diff
@@ -0,0 +1,302 @@
+export type PropertyFlags = StructFlags;
+export type MethodFlags = { static?: boolean };
+export type DeclarationFlag = { final?: boolean };
+export type Kind = 'class' | 'struct' | 'extension' | 'enum';
+export type VariableDeclaration = {
+  value: string | undefined;
+  name: string;
+  type: string;
+  flags: VariableFlags;
+};
+export type StructProperty = VariableDeclaration & {
+  access: Access;
+  comment?: string;
+  getter?: string;
+  setter?: string;
+  flags: PropertyFlags;
+};
+export type MethodArgument = VariableDeclaration & {
+  flags: PropertyFlags;
+};
+export type StructMethod = {
+  args: MethodArgument[];
+  implementation: string;
+  name: string;
+  access: Access;
+  returnType: string | null;
+  flags: MethodFlags;
+  comment: string;
+};
+export class SwiftDeclarationBlock {
+  _name: string = '';
+  _kind: Kind = 'struct';
+  _protocols: string[] = [];
+  _access: Access = 'DEFAULT';
+  _properties: StructProperty[] = [];
+  _methods: StructMethod[] = [];
+  _comment: string = '';
+  _block: string[] = [];
+  _enumValues: { [name: string]: string } = {};
+  _flags: DeclarationFlag = { final: false };
+
+  access(access: Access): SwiftDeclarationBlock {
+    this._access = access;
+
+    return this;
+  }
+
+  final(isFinal: boolean = true): SwiftDeclarationBlock {
+    this._flags.final = isFinal;
+    return this;
+  }
+
+  withComment(comment: string | StringValueNode | null): SwiftDeclarationBlock {
+    if (comment) {
+      this._comment = transformComment(comment, 0);
+    }
+    return this;
+  }
+
+  withName(name: string | NameNode): SwiftDeclarationBlock {
+    this._name = typeof name === 'object' ? (name as NameNode).value : name;
+
+    return this;
+  }
+
+  withBlock(block: string): SwiftDeclarationBlock {
+    this._block = [block];
+
+    return this;
+  }
+  appendBlock(block: string): SwiftDeclarationBlock {
+    this._block.push(block);
+    return this;
+  }
+
+  asKind(kind: Kind): SwiftDeclarationBlock {
+    this._kind = kind;
+    return this;
+  }
+
+  addProperty(
+    name: string,
+    type: string,
+    value?: string,
+    access: Access = 'public',
+    flags: PropertyFlags = {},
+    comment?: string,
+    getter?: string,
+    setter?: string
+  ): SwiftDeclarationBlock {
+    this._properties.push({
+      name,
+      type,
+      value,
+      access,
+      flags: {
+        optional: false,
+        ...flags,
+      },
+      comment,
+      getter,
+      setter,
+    });
+
+    return this;
+  }
+
+  withProtocols(protocols: string[]): SwiftDeclarationBlock {
+    this._protocols = protocols;
+    return this;
+  }
+
+  addEnumValue(name: string, value?: string): SwiftDeclarationBlock {
+    if (this._kind !== 'enum') {
+      throw new Error(`Can not add enum values for block type ${this._kind}`);
+    }
+
+    this._enumValues[name] = value || name;
+    return this;
+  }
+
+  addClassMethod(
+    name: string,
+    returnType: string | null,
+    impl: string,
```

Comment:
Is there a reason why it is abbreviated?

Reply:
not really. Wanted to type less